rfc/helm.md

```diff
@@ -0,0 +1,146 @@
+Helm Chart Support
+====================
+
+The current CLI install process uses Helm libraries under the hood, but just as a template library. There is currently a Helm chart under the `charts` directory that allows installing the control plane, but uninjected, and there's no `values.yaml` file provided.
+                                                                                                                                                                                                                                             
+The intention here is to provide a new chart with an injected control plane so that users can install linkerd through a simple `helm install incubator/linkerd2` command.
+                                                                                                                                                                                                                                             
+Helm charts can't rely on go code besides the functions provided by the go template library, the Sprig library, and a few extra functions provided by Helm itself. This implies a few compromises:
+
+- We can't have the same level of validation that `linkerd install` currently provides. Some of the basic checks can be performed and have `helm install` fail using [Sprig's fail](http://masterminds.github.io/sprig/flow_control.html). The other checks can be performed with a new command `linkerd check --chart values.yaml`.
+- We can't validate the install options provided in `values.yaml`. Instead, a new set of `linkerd check` checks could help catching invalid options, post-install.
+- We should provide a comprehensive `values.yaml` file that would contain the most common settings, but heavily annotated to instruct users about alternate settings for advanced scenarios.    
+- Other alternative `values.yaml` files should be available, for different configuration scenarios like HA for example.
+- Helm's crypto functions only allow us to use RSA certs/keys. We can move the cert/keys from the webhook configs (`proxy-injector` and `sp-validator`) to use RSA. As for the trust root for identity, we decided the user should provide their own in `values.yaml`. The docs should have instructions on how to generate that cert/key.
```

Comment:
Any reasons why we can't let users bring their own CA? It will get persisted in the `Secret`s across upgrades anyway.

Reply:
You mean trust root? I thought this was saying it is required that users bring their own trust root as part of the helm chart install.

Reply:
Sorry, I was referring to the webhooks' CA bundle/trust, which is used to secured communication between the k8s API server and the webhooks.

> We can move the cert/keys from the webhook configs (`proxy-injector` and `sp-validator`) to use RSA.

:point_up: I take this to mean we are using Helm's functions to generate the CA bundle/trust for the webhooks.

Reply:
Let's just do it ourselves for v1 (and not allow folks to override). If it is a customer request, it wouldn't be hard to change.